internal/cmd/commands/connect/connect.go

```diff
@@ -732,3 +739,12 @@ func (c *Command) handleExec(clientProxy *apiproxy.ClientProxy, passthroughArgs
 	}
 	c.execCmdReturnValue.Store(0)
 }
+
+func isPortAvailable(listenPort int64, listenAddr string) bool {
+	listener, err := net.Listen("tcp", fmt.Sprintf("%s:%d", listenAddr, listenPort))
+	if err != nil {
+		return false
+	}
+	_ = listener.Close()
```

Comment:
Try to handle error closing  
```suggestion
defer func() {
		err = listener.Close()
		if err != nil {
                         error message
		}
	}()
```